lib/bn.js

```diff
@@ -3001,6 +3001,9 @@
     }
     input.words[i - 10] = prev >>> 22;
     input.length -= 9;
+    if (input.length > 1 && input.words[input.length - 1] === 0) {
```

Comment:
And check just `prev === 0` here. Do you think that `length` could be `<= 1` here?